File: myadmin-service-api/src/main/java/za/ac/unisa/myadmin/exam/service/ExamAdmissionService.java
package za.ac.unisa.myadmin.exam.service;

import java.util.List;

import za.ac.unisa.myadmin.common.exceptions.InvalidParameterException;
import za.ac.unisa.myadmin.common.exceptions.MissingParameterException;
import za.ac.unisa.myadmin.common.exceptions.OperationFailedException;
import za.ac.unisa.myadmin.exam.dto.ExamAdmissionInfo;

public interface ExamAdmissionService {

	public List<ExamAdmissionInfo> getExamAdmissions() throws OperationFailedException;

	public List<ExamAdmissionInfo> getExamAdmissionsByYear(Integer year)
			throws MissingParameterException, InvalidParameterException, OperationFailedException;

	public List<ExamAdmissionInfo> getExamAdmissionsByExamPeriodCode(Integer examPeriodCode)
			throws MissingParameterException, InvalidParameterException, OperationFailedException;

	public List<ExamAdmissionInfo> getExamAdmissionsByExamType(Integer examType)
			throws MissingParameterException, InvalidParameterException, OperationFailedException;

	public List<ExamAdmissionInfo> getExamAdmissionsByYearAndExamPeriodCode(Integer year, Integer examPeriodCode)
			throws MissingParameterException, InvalidParameterException, OperationFailedException;

	public List<ExamAdmissionInfo> getExamAdmissionsByYearAndExamType(Integer year, Integer examType)
			throws MissingParameterException, InvalidParameterException, OperationFailedException;

	public List<ExamAdmissionInfo> getExamAdmissionsByExamPeriodCodeAndExamType(Integer examPeriodCode,
			Integer examType)
			throws MissingParameterException, InvalidParameterException, OperationFailedException;

	public List<ExamAdmissionInfo> getExamAdmissionsByYearAndExamPeriodCodeAndExamType(Integer year,
			Integer examPeriodCode, Integer examType)
			throws MissingParameterException, InvalidParameterException, OperationFailedException;

}

File: myadmin-angular/projects/exam-timetable-app/src/app/services/exam-admission.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

import { ExamAdmissionInfo } from '../info-objects';

@Injectable()
export class ExamAdmissionService {

  constructor(private http: HttpClient) {
  }

  getExamAdmissions(): Observable<ExamAdmissionInfo[]> {
    return this.http.get<ExamAdmissionInfo[]>('/myadmin-exam-services/examservices/examadmissions');
  }

  getExamAdmissionsByYear(year: number): Observable<ExamAdmissionInfo[]> {
    const params = new HttpParams().set('year', <string><any>year);

    return this.http.get<ExamAdmissionInfo[]>('/myadmin-exam-services/examservices/examadmissions', { params });
  }

  getExamAdmissionsByExamPeriodCode(examPeriodCode: number): Observable<ExamAdmissionInfo[]> {
    const params = new HttpParams().set('examPeriodCode', <string><any>examPeriodCode);

    return this.http.get<ExamAdmissionInfo[]>('/myadmin-exam-services/examservices/examadmissions', { params });
  }

  getExamAdmissionsByExamType(examType: number): Observable<ExamAdmissionInfo[]> {
    const params = new HttpParams().set('examType', <string><any>examType);

    return this.http.get<ExamAdmissionInfo[]>('/myadmin-exam-services/examservices/examadmissions', { params });
  }

  getExamAdmissionsByYearAndExamPeriodCode(year: number, examPeriodCode: number): Observable<ExamAdmissionInfo[]> {
    const params = new HttpParams()
      .set('year', <string><any>year)
      .set('examPeriodCode', <string><any>examPeriodCode);

    return this.http.get<ExamAdmissionInfo[]>('/myadmin-exam-services/examservices/examadmissions', { params });
  }

  getExamAdmissionsByYearAndExamType(year: number, examType: number): Observable<ExamAdmissionInfo[]> {
    const params = new HttpParams()
      .set('year', <string><any>year)
      .set('examType', <string><any>examType);

    return this.http.get<ExamAdmissionInfo[]>('/myadmin-exam-services/examservices/examadmissions', { params });
  }

  getExamAdmissionsByExamPeriodCodeAndExamType(examPeriodCode: number, examType: number): Observable<ExamAdmissionInfo[]> {
    const params = new HttpParams()
      .set('examPeriodCode', <string><any>examPeriodCode)
      .set('examType', <string><any>examType);

    return this.http.get<ExamAdmissionInfo[]>('/myadmin-exam-services/examservices/examadmissions', { params });
  }

  getExamAdmissionsByYearAndExamPeriodCodeAndExamType(year: number, examPeriodCode: number, examType: number): Observable<ExamAdmissionInfo[]> {
    const params = new HttpParams()
      .set('year', <string><any>year)
      .set('examPeriodCode', <string><any>examPeriodCode)
      .set('examType', <string><any>examType);

    return this.http.get<ExamAdmissionInfo[]>('/myadmin-exam-services/examservices/examadmissions', { params });
  }
}
